src/Builder/Encoder/CombinedFieldQueryEncoder.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder\Encoder;
+
+use LogicException;
+use MongoDB\Builder\Type\CombinedFieldQuery;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Exception\UnsupportedValueException;
+use stdClass;
+
+use function get_debug_type;
+use function get_object_vars;
+use function is_array;
+use function is_object;
+use function sprintf;
+
+/** @template-extends AbstractExpressionEncoder<stdClass, CombinedFieldQuery> */
+class CombinedFieldQueryEncoder extends AbstractExpressionEncoder
+{
+    /** @template-use EncodeIfSupported<stdClass, CombinedFieldQuery> */
+    use EncodeIfSupported;
+
+    public function canEncode(mixed $value): bool
+    {
+        return $value instanceof CombinedFieldQuery;
+    }
+
+    public function encode(mixed $value): stdClass
```

Comment:
Noted that this was lifted from `BuilderEncoder::encodeCombinedFilter()`, so I'm not going to spend time reviewing this (and similar) methods.